llms/mlx_lm/tuner/utils.py

```diff
@@ -58,11 +59,17 @@ def linear_to_lora_layers(
             f"Requested {num_lora_layers} LoRA layers "
             f"but the model only has {num_layers} layers."
         )
-    cls = DoRALinear if use_dora else LoRALinear
 
-    def to_lora(lin):
-        return cls.from_linear(
-            lin,
+    def to_lora(layer):
+        if isinstance(layer, (nn.Linear, nn.QuantizedLinear)):
+            LoRALayer = DoRALinear if use_dora else LoRALinear
+        elif isinstance(layer, (SwitchLinear, QuantizedSwitchLinear)):
+            LoRALayer = LoRASwitchLinear
```

Comment:
Should we add `use_dora` to the condition here? Or throw if trying to DoRA-fy a switch linear layer (until we support Dora)?